.github/workflows/build.yml

```diff
@@ -162,13 +162,23 @@ jobs:
           retry_wait_seconds: 10
           command: docker pull docker.pkg.github.com/apache/incubator-nuttx-testing/nuttx-ci-linux
 
+      - name: Export NuttX Repo SHA
+        run:  echo "::set-env name=nuttx_sha::`git -C sources/nuttx rev-parse HEAD`"
+      - name: Refresh Git Credentials
+        uses: actions/checkout@v2
+        with:
+          repository: apache/incubator-nuttx
```

Comment:
Does this mean we fetch the source code again?

Reply:
It cleans the repo and resets all the refs (no way to avoid it, I opened a ticket on the checkout action), but we still have all the objects cached in `.git`

Reply:
The current method require each job(20~):
1.Clean the repo and reset all refs
2.Fetch the tag from server
Can we save these two step if we move the tag fetch to 'Fetch-Source->Checkout nuttx repo'? I am wondering which method is more fast and stable.

Reply:
I am testing a build where we just pass the refs